### ISSUE TITLE
Some ERRORS about regexp_funtion

### ISSUE THREAD

**Issue by leolong0325** [orafce/orafce#246]
Hi, recently I was using some regular functions in the orafce plug-in，I found that the following case is inconsistent with oracle：
```
test_oracle=# select REGEXP_COUNT('a,b' || chr(10) || 'c','[^,]+') from dual;
 regexp_count 
--------------
            3
(1 row)
```
The result in the Oracle database is：
```
SQL> select REGEXP_COUNT('a,b' || chr(10) || 'c','[^,]+') from dual;

REGEXP_COUNT('A,B'||CHR(10)||'C','[^,]+')
-----------------------------------------
					2
```
Then I looked at the code of the plug-in and found that it was the effect of the modifier 'p'
![image](https://github.com/orafce/orafce/assets/97018470/cc53951a-9f13-442f-947f-128b575508f5)
I checked the regexp.c file and found 'p' means ~Perl, \n affects . [^：
![image](https://github.com/orafce/orafce/assets/97018470/cf982f3f-5e9c-4ed4-91a1-5b9d4c965daf)
It can be seen that the 'p' modifier affects the matching of [^ to \n, making the result wrong. The following case can prove this.
```
test_oracle=# select REGEXP_COUNT('a,b' || chr(10) || 'c','b.c') from dual;
 regexp_count 
--------------
            0
(1 row)
```
Other regular functions also have similar problems：
```
test_oracle=# SELECT regexp_substr('a,b
c','[^,]+',1,2) FROM DUAL;
 regexp_substr 
---------------
 b
(1 row)

SQL> SELECT regexp_substr('a,b
c','[^,]+',1,2) FROM DUAL;

REG
---
b
c

```
When I tried to remove the 'p' modifier, I found that this case passed, but other cases had problems：
```
test_oracle=# SELECT regexp_substr('a,b
c','[^,]+',1,2) FROM DUAL;
 regexp_substr 
---------------
 b            +
 c
(1 row)

test_oracle=# SELECT regexp_substr('a,b
c','.',1,4) FROM DUAL;
 regexp_substr 
---------------
              +
 
(1 row)

SQL> SELECT regexp_substr('a,b
  2  c','.',1,4) FROM DUAL;

R
-
c

```

So I want to know how I can modify it so that the first case result is correct.

Thank you and looking forward to your reply!

**Comment by okbob** [comment]
This is not my code, and I don't know the Oracle's regexp too much. Looks so Oracle ignore other rows. I have to study this issue more deep

**Comment by darold** [comment]
@okbob, if you want and to save your time, I could have a look on this next week.

**Comment by okbob** [comment]
pá 8. 12. 2023 v 9:39 odesílatel Gilles Darold ***@***.***>
napsal:

> @okbob <https://github.com/okbob>, if you want and to save your time, I
> could have a look on this next week.
>

please, if it is possible.

**Comment by darold** [comment]
I have founded a partial solution. Using modifier `sg` instead of `pg` fixes all the above examples. There is a problem remaining with the following test:
```
gilles=#  select oracle.REGEXP_COUNT('a,b' || chr(10) || 'c','b.c');
 regexp_count 
--------------
            1
(1 line)

SQL>  select REGEXP_COUNT('a,b' || chr(10) || 'c','b.c') from dual;

REGEXP_COUNT('A,B'||CHR(10)||'C','B.C')
---------------------------------------
				      0
```

PS: the `p` modifier doesn't mean Perl, it is for: "partial newline-sensitive matching"

**Comment by darold** [comment]
A possible solution could be to replace any occurrence of `.` by `[^\n]`

**Comment by okbob** [comment]
should be fixed in 4.9.0 release
